src/host/PtySignalInputThread.cpp

```diff
@@ -227,14 +227,6 @@ void PtySignalInputThread::_Shutdown()
     // Trigger process shutdown.
     CloseConsoleProcessState();
 
-    // If we haven't terminated by now, that's because there's a client that's still attached.
-    // Force the handling of the control events by the attached clients.
-    // As of MSFT:19419231, CloseConsoleProcessState will make sure this
-    //      happens if this method is called outside of lock, but if we're
-    //      currently locked, we want to make sure ctrl events are handled
-    //      _before_ we RundownAndExit.
-    ProcessCtrlEvents();

```

Comment:
Why do we not have to do this anymore? IIRC there's a very bizarre edge case bug that can occur if we don't dispatch ctrl events before tearing down

Reply:
I believe this might be a copy-paste mistake...
This function is never called while the console is locked (see `_GetData` in the same file), so this can't possibly happen.
However I've just pushed a change which makes the entire code a lot more robust in general in my opinion. 🙂

Reply:
I'm definitely worried about making changes here. If it's a copy/paste error, we should make sure we understand when it was introduced and when it became an error. Pull up the team OneNote and use the guidance on importing pre-github history if you need to! :)

Reply:
I found the original comments on the PR that added this function (!2698602):
> Yeah, my point was that the two functions (`PtySignalInputThread::_Shutdown` and `VtIo::_ShutdownIfNeeded`) should probably have the same signature, one way or the other, because you fixed the jiggling the handle thing. Either one of them is double-processing ctrl events or the other is not processing them at all. They should be consistent.
> [...]
> Derp yea, actually I believe `_ShutdownIfNeeded` is the right one. For both of these, if they're called under lock, we don't want to `RundownAndExit` before `ProcessCtrlEvents` is called.

This issue has now been resolved by jiggling in a different way instead. 😅